h5py/tests/old/test_h5d_direct_chunk.py

```diff
@@ -0,0 +1,92 @@
+            a = numpy.random.rand(100, 100).astype('float32')
+            dataset.id.write_direct_chunk((index, 0, 0), a.tostring(), filter_mask=1)
+            array[index] = a
+
+        filehandle.close()
+
+        # checking
+        filehandle = h5py.File(filename, "r")
+        for i in range(10):
+            read_data = filehandle["data"][i]
+            self.assertTrue((array[i] == read_data).all())
+
+
+@ut.skipUnless(h5py.version.hdf5_version_tuple >= (1, 10, 2), 'Direct Chunk Reading requires HDF5 >= 1.10.2')
+class TestReadDirectChunk(TestCase):
+    def test_read_compressed_offsets(self):
+
+        filename = self.mktemp().encode()
+        filehandle = h5py.File(filename, "w")
+
+        frame = numpy.arange(16).reshape(4, 4)
+        frame_dataset = filehandle.create_dataset("frame",
+                                                  data=frame,
+                                                  compression="gzip",
+                                                  compression_opts=9)
+        dataset = filehandle.create_dataset("compressed_chunked",
+                                            data=[frame, frame, frame],
+                                            compression="gzip",
+                                            compression_opts=9,
+                                            chunks=(1, ) + frame.shape)
+        filter_mask, compressed_frame = frame_dataset.id.read_direct_chunk((0, 0))
+        # No filter must be disabled
+        self.assertEqual(filter_mask, 0)
+
+        for i in range(dataset.shape[0]):
+            filter_mask, data = dataset.id.read_direct_chunk((i, 0, 0))
+            self.assertEqual(compressed_frame, data)
+            # No filter must be disabled
+            self.assertEqual(filter_mask, 0)
+
+    def test_read_write_chunk(self):
+
+        filename = self.mktemp().encode()
+        filehandle = h5py.File(filename, "w")
+
+        # create a reference
+        frame = numpy.arange(16).reshape(4, 4)
+        frame_dataset = filehandle.create_dataset("source",
+                                                  data=frame,
+                                                  compression="gzip",
+                                                  compression_opts=9)
+        # configure an empty dataset
+        filter_mask, compressed_frame = frame_dataset.id.read_direct_chunk((0, 0))
+        dataset = filehandle.create_dataset("created",
+                                            shape=frame_dataset.shape,
+                                            maxshape=frame_dataset.shape,
+                                            chunks=frame_dataset.chunks,
+                                            dtype=frame_dataset.dtype,
+                                            compression="gzip",
+                                            compression_opts=9)
+
+        # copy the data
+        dataset.id.write_direct_chunk((0, 0), compressed_frame, filter_mask=filter_mask)
```

Comment:
I'd like a test which uses a dataset with compression defined and writes a chunk with the compression filter *disabled*. It should then check that:

- Accessing the relevant data normally (`ds[i]`) gives the expected data
- Reading the chunk through `read_direct_chunk` gives us the same filter_mask back.

Reply:
Oh I see you already tried to do a test like this and described an error above. I'm not sure what that could mean.

The [example here](https://confluence.hdfgroup.org/display/HDF5/Using+the+Direct+Chunk+Write+Function) only uses `filter_mask = 0x00000001` for writing uncompressed data. I wonder if claiming to disable more filters than the dataset has does something strange, or if there's an edge case with setting 0xffffffff.

Reply:
Ill try that. But i think it was not the pbm.
I guess the problem is the use of dataset which can grow.